src/careamics/cli/conf.py

```diff
@@ -0,0 +1,404 @@
+"""Configuration building convinience functions for the CAREamics CLI."""
+
+import os
+import sys
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Tuple
+
+import click
+import typer
+import yaml
+from typing_extensions import Annotated
+
+# import hydra
+from ..config import (
+    Configuration,
+    create_care_configuration,
+    create_n2n_configuration,
+    create_n2v_configuration,
+)
+
+WORK_DIR = Path(os.getcwd())
+
+app = typer.Typer()
+
+
+def _save_config(config: Configuration, fp: Path) -> None:
+    """
+    Function to save careamics Configuration object to a yaml file.
+
+    Parameters
+    ----------
+    config : Configuration
+        CAREamics configuration object.
+    fp : Path
+        File path to save to.
+    """
+    with open(fp, "w") as file:
+        yaml.dump(config.model_dump(), file, indent=2)
+
+
+def _config_builder_exit(ctx: typer.Context, config: Configuration) -> None:
+    """
+    Function to be called at the end of a CLI configuration builder.
+
+    Saves the `config` object and performs other functionality depending on the command
+    context.
+
+    Parameters
+    ----------
+    ctx : typer.Context
+        Typer Context.
+    config : Configuration
+        CAREamics configuration.
+    """
+    conf_path = (ctx.obj.dir / ctx.obj.name).with_suffix(".yaml")
+    _save_config(config, conf_path)
+    if ctx.obj.print:
+        print(yaml.dump(config.model_dump(), indent=2))
+
+
+@dataclass
+class ConfOptions:
+    """Data class for containing CLI `conf` command option values."""
+
+    dir: Path
+    name: str
+    force: bool
+    print: bool
+
+
+@app.callback()
+def conf_options(  # numpydoc ignore=PR01
```

Comment:
Is the `numpydoc ignore` because it is WIP? Or another reason?

Reply:
PR01 is the error that the parameters have not been defined in the docstrings. However, Typer takes the docstrings and adds it to the help message, and it looks a mess with the parameter descriptions because Typer seems to remove newlines. Additionally, the parameters are already defined with the help argument in the typer.Option objects.

Reply:
makes sense!